src/Classes/PolygonMask.ts

```diff
@@ -0,0 +1,222 @@
+                return;
+            }
+        }
+
+        this.width = parseFloat(this.svg.getAttribute("width") || "0");
+        this.height = parseFloat(this.svg.getAttribute("height") || "0");
+
+        /* centering of the polygon mask */
+        this.offset = {
+            x: container.canvas.width / 2 - this.width / 2,
+            y: container.canvas.height / 2 - this.height / 2
+        };
+
+
+        const len = this.path.pathSegList.numberOfItems;
+        const polygonRaw = [];
+        const p = {
+            x: 0,
+            y: 0
+        };
+
+        for (let i = 0; i < len; i++) {
+            const pathSeg: any = this.path.pathSegList.getItem(i);
+
+            switch (pathSeg.pathSegType) {
+                //
+                // Absolute
+                //
+                case window.SVGPathSeg.PATHSEG_MOVETO_ABS:
+                case window.SVGPathSeg.PATHSEG_LINETO_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_ABS:
+                case window.SVGPathSeg.PATHSEG_ARC_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_SMOOTH_ABS:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_SMOOTH_ABS:
+                    p.x = pathSeg.x;
+                    p.y = pathSeg.y;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_HORIZONTAL_ABS:
+                    p.x = pathSeg.x;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_VERTICAL_ABS:
+                    p.y = pathSeg.y;
+                    break;
+
+                //
+                // Relative
+                //
+                case window.SVGPathSeg.PATHSEG_LINETO_REL:
+                case window.SVGPathSeg.PATHSEG_MOVETO_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_REL:
+                case window.SVGPathSeg.PATHSEG_ARC_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_CUBIC_SMOOTH_REL:
+                case window.SVGPathSeg.PATHSEG_CURVETO_QUADRATIC_SMOOTH_REL:
+                    p.x += pathSeg.x;
+                    p.y += pathSeg.y;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_LINETO_HORIZONTAL_REL:
+                    p.x += pathSeg.x;
+                    break;
+                case window.SVGPathSeg.PATHSEG_LINETO_VERTICAL_REL:
+                    p.y += pathSeg.y;
+                    break;
+
+                case window.SVGPathSeg.PATHSEG_UNKNOWN:
+                case window.SVGPathSeg.PATHSEG_CLOSEPATH:
+                    continue; // Skip the closing path (and the UNKNOWN)
+                    break;
+            }
+
+            polygonRaw.push([p.x + this.offset.x, p.y + this.offset.y]);
+        }
+
+        return polygonRaw;
+    }
+
+    drawDebugPolygon() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class method 'drawDebugPolygon' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5042590)